### PR TITLE
Make pod status to Running if some are Running and some are Completed

### LINKED ISSUE #62588: Bug: `Kubectl get pod` - Pod is running and has two Containers. Container 1 exits with success.
**Is this a BUG REPORT or FEATURE REQUEST?**:
> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:
When one pod is running and has two Containers. Container 1 exits with success and container 2 is running.
`kubectl get test -o=json` gives correct info -  the pod status is `Running`. 
However, `kubectl get pod test -o=wide`, the pod status is `Completed`.

**What you expected to happen**:
`kubectl get pod test -o=wide`, the pod status is `Running`.

**How to reproduce it (as minimally and precisely as possible)**:
```
apiVersion: v1
kind: Pod
metadata:
  name: test
  labels:
    name: test
spec:
  restartPolicy: Never
  containers:
  - name: test1
    image: busybox
    command: ["sh", "-c", "echo hello"]
  - name: test2
    image: busybox
    command: ["sh", "-c", "sleep 10000"]
```

**Anything else we need to know?**:

**Environment**: 
- Kubernetes version (use `kubectl version`): 1.9.3
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by ceshihao**
**What this PR does / why we need it**:
Make pod status to Running if some are Running and some are Completed

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #62588

**Special notes for your reviewer**:
Only Set Pod status to Completed when no other reason, no Running container and only Completed containers.

Set status to Running if some are Running and some are Completed

**Release note**:

```release-note
Set pod status to "Running" if there is at least one container still reporting as "Running" status and others are "Completed".
```

### CODE REVIEW

**Comment by hochuenw** [comment]
@ceshiha, it seems I can not ping a pod which some containers are Running and some are Completed through service. Can this printer fix also fix that issue?

**Comment by ceshihao** [comment]
@hochuenw 

No, it is different. I think it is another issue.

**Review by dixudx — COMMENTED** [review]
Better add a test for this change.

File: pkg/printers/internalversion/printers.go
Comment (juanvallejo) [inline pkg/printers/internalversion/printers.go]: What if the pod's status is `Completed`? 
This line would make it so that Completed pods would appear with a `Succeeded` status, rather than `Completed`.

Why not simplify to this (or something similar)?

```go
...
if !initializing {
    restarts = 0
    hasRunning := false
    for i := ... {
        ...
        // last "else if" in loop 
        } else if container.Ready && container.State.Running != nil {
            hasRunning = true
            readyContainers++
        }
    }

    // revert status back to "Running" if there is at least onr
    // container still reporting the "Running" status
    if reason == "Completed" && hasRunning {
        reason = "Running"
    }
}
```

**Comment by juanvallejo** [comment]
@soltysh fyi

File: pkg/printers/internalversion/printers.go
Comment (ceshihao) [reply pkg/printers/internalversion/printers.go]: Thank you. 

Sounds great.
Change the pod status codes according to your comment.

**Comment by mengqiy** [comment]
We probably want a release note, since this is a behavior change to fix a bug.
The code LGTM

**Review by juanvallejo — APPROVED** [review]

**Comment by juanvallejo** [comment]
lgtm pending a release note

**Comment by ceshihao** [comment]
Add a meaningful release note.

**Comment by dixudx** [comment]
@ceshihao Would you please squash your commits. No need to have 4 commits on 2 files.

**Comment by ceshihao** [comment]
@dixudx 
Squash to a single commit.

**Comment by dixudx** [comment]
ping @kubernetes/sig-cli-maintainers for approval.

**Pull request merged**
